JSONText(string) for embedding raw JSON in the output

stevengj opened the pull request:
This exports a `JSONText(s::String)` object that you can use to embed raw JSON in directly in the output.

For example, this would be useful in IJulia for JuliaLang/IJulia.jl#621

stevengj commented:
Unrelated CI failure on 0.7 (same as `master`) due to `OffsetArray` problem.

TotalVerb commented:
This will ignore the `indent=x` keyword argument for `json`, so I'm not 100% comfortable with it, as-is. Can we "enforce" the "no-extra whitespace" rule in the docstring (i.e. document that the JSON should be ensured to be valid and formatted correctly), and reparse and print for general `::SC` (only specializing on `::CompactContext`), so that users will not ever get unexpected results?

stevengj commented:
I updated the PR to document that valid JSON is required.  Also, as you suggest, I updated it to re-parse and prints for general contexts, only printing as-is for `CompactContext`.
